### DISCUSSION TITLE
Decentralization?

### DISCUSSION THREAD

**Discussion by ureru** [stoatchat/discussions#249] [Q&A] [answered]
Will Revolt ever be decentralized? Will it ever be able to support making a server completely P2P?

**Comment by Zomatree** [comment, marked as answer]
No, https://developers.revolt.chat/faq/federation

**Reply by ureru** [reply]
I'm not sure that answers the question. Federation is only one type of decentralization. Or does that mean that P2P servers won't happen?

**Reply by insertish** [reply]
What do you mean by a P2P server? How would that work?

**Reply by ureru** [reply]
The way that I imagine it is that each peer keeps a copy of the server's messages and settings and stuff and a list of the users on the server along with their IPs, and if something on the server changes then all of the clients try to update their data, and by connecting to any other peer who's currently online they can connect to the server, hopefully an up to date version. This has other benefits than just privacy and security as well, although it's also a strong benefit from those points (as long as you can trust your peers, but only your peers should have access to their e2ee keys so there's that). But the server will never be deleted, and I'm aware that the people in charge of regulating what servers can be on Revolt may believe that their decisions for what servers to delete are always just or whatever but after being on discord and being threatened with server deletion by discord staff over nothing, I do not like to deal with the possibility of server deletion anymore. But really, most of the concern is privacy and security.

Also, this is not only a question about P2P servers, just like it's not only about federation, this is a question about decentralization in general. About federation, as I understand it that's making a whole interconnected large scale network of different servers, right? I don't see why that has to be necessary. So is there any type of decentralization I could expect or hope for? Even if it's just allowing people to host their own servers that have no interconnection with other servers. Or some other option I didn't think of.

Decentralization basically just means everything is not in the hands of Revolt's server(s) somehow. There are many different ways to accomplish this.

Actually, while I'm at it, does anyone know of a chat program that does offer P2P communities?

**Reply by BitForger** [reply]
I understand that you aren't in the business to compete with matrix, but right now there is a need for a platform that is privacy-centric but provides the same (even better) feature set and user experience than Discord. I understand that's what you're doing, and while it's a step forward, there is a degree of trust that has to be placed in the owner of the project that you will uphold your end of the bargain.
I would strongly encourage an approach similar to matrix or some way that allows a user to create their own communities while keeping data in their control (especially in the case of their servers/spaces)
